source/comp/markv.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// MARK-V is a compression format for SPIR-V binaries. It strips away
+// non-essential information (such as result ids which can be regenerated) and
+// uses various bit reduction techiniques to reduce the size of the binary and
+// make it more similar to other compressed SPIR-V files to further improve
+// compression of the dataset.
+
+#ifndef SPIRV_TOOLS_MARKV_HPP_
+#define SPIRV_TOOLS_MARKV_HPP_
+
+#include <string>
+#include <vector>
+
+#include "markv_model.h"
+#include "spirv-tools/libspirv.h"
+
+namespace spvtools {
+
+struct markv_encoder_options_t {
```

Comment:
Prefer CamelCase with leading capital for C++ types.

I usually prefer an opaque options object.   I guess this is ok provided all future additions to the struct get member initializers.  Otherwise it will be too easy for clients to only explicitly initialize some;  and without a member initializer you'll get default zero-initialized (essentially).  But it's a sharp edge that doesn't have to remain.

(Also with an opaque object the setter methods can return a reference to the options object, and then clients can chain setters.  It's a nice pattern.)